lib/matplotlib/backends/backend_ps.py

```diff
@@ -952,8 +977,36 @@ def print_figure_impl(fh):
                         fh.write(_font_to_ps_type3(font_path, glyph_ids))
                     else:
                         try:
-                            convert_ttf_to_ps(os.fsencode(font_path),
-                                              fh, fonttype, glyph_ids)
+                            _log.debug(
+                                f"SUBSET {font_path} characters: "
+                                f"{''.join(chr(c) for c in chars)}"
+                            )
+                            fontdata = getSubset(
+                                font_path, "".join(chr(c) for c in chars)
+                            )
+                            _log.debug(
+                                f"SUBSET {font_path} "
+                                f"{os.stat(font_path).st_size} "
+                                f"↦ {len(fontdata)}"
+                            )
+
+                            # give ttconv a subsetted font
+                            # along with updated glyph_ids
+                            with tempfile.NamedTemporaryFile(
+                                suffix=".ttf"
+                            ) as tmp:
```

Comment:
as above re: intermediate file.

Reply:
`convert_ttf_to_ps` takes a filepath as the first parameter, because read_font takes fontpath too:
https://github.com/matplotlib/matplotlib/blob/d1567715f1913ce935462c408f02e70e6299b6cb/extern/ttconv/pprdrv_tt.cpp#L1223

for PDF backend we're not using this function, but here I think we need to? (unless there's a way around it)

Reply:
I guess it may be worth seeing whether ttconv can be fixed to support filelikes as well, but this can be done later.

Reply:
`read_font` it's being called from the below function (which is called by `convert_ttf_to_ps`): https://github.com/matplotlib/matplotlib/blob/d1567715f1913ce935462c408f02e70e6299b6cb/src/_ttconv.cpp#L132

Reply:
> I guess it may be worth seeing whether ttconv can be fixed to support filelikes as well, but this can be done later.

I added a TODO just before it